### PR TITLE
Refactor UI publishing and polish tests

### PR DESCRIPTION
**Pull request by roidelapluie**
Signed-off-by: Julien Pivotto <roidelapluie@o11y.eu>

### CODE REVIEW

**Comment by Nexucis** [comment]
looks good, from my point of view `test.yaml` is not an accurate name as it is publishing also.
Why not splitting the publish job and the tests in two separate files ?

**Comment by roidelapluie** [comment]
Because of the required relationship between the jobs.

**Comment by Nexucis** [comment]
ah yeah ok.

Why not renaming the file then ?

**Comment by roidelapluie** [comment]
I have renamed it to CI

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -119,11 +126,29 @@ jobs:
...
+        if: github.repository == 'prometheus/snmp_exporter'
+      - name: Lint
+        uses: golangci/golangci-lint-action@v3.2.0
+        with:
+          version: v1.45.2
 
   publish_main:
+    name: Publish main branch artefacts
     runs-on: ubuntu-latest
     needs: [test_ui, test_go, test_windows, build_all]
```
Comment (Nexucis) [inline .github/workflows/CI.yml:150]: do you think this job should depend on `golangci` as well ?

**Comment by roidelapluie** [comment]
Alright, more dependencies :)

**Pull request merged**